### PR TITLE
Fix the DCDO and Gatekeeper loaders for dev environment

### PR DESCRIPTION
**Pull request by trevorb**

### CODE REVIEW

File: bin/dynamic_config/update_dcdo.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+SCRIPT=$(readlink -f $0)
+SCRIPT_PATH=`dirname $SCRIPT`
+FULL_CONFIG_PATH=`readlink -f "$1"`
+
+cd $SCRIPT_PATH/../..
+bundle exec ruby $SCRIPT_PATH/lib/update_dcdo.rb $FULL_CONFIG_PATH
```
Comment (wjordan) [inline bin/dynamic_config/update_dcdo.sh:7]: It seems like all this Bash script is doing is executing the ruby script with a single argument. Is there an issue with setting the Ruby script executable and invoking it directly (the pattern used by all the other /bin scripts in this folder)?

File: bin/dynamic_config/lib/update_dcdo.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env ruby
+
+puts "Loading environment..."
+require_relative '../../../dashboard/config/environment'
+require 'dynamic_config/loaders/dcdo_loader'
+
+puts "Updating config..."
+DCDOLoader.load(ARGV[0])
+puts "Done"
```
Comment (wjordan) [inline bin/dynamic_config/lib/update_dcdo.rb:9]: Can we document that this script requires a single argument to a yaml file containing local DCDO configuration, or at least set an informative error message when invoked without one? This is what happens when I invoke without any arguments:

```
$ ./update_dcdo.sh 
Loading environment...
Updating config...
/home/will/Code/code-dot-org/lib/dynamic_config/loaders/dcdo_loader.rb:52:in `initialize': no implicit conversion of nil into String (TypeError)
    from /home/will/Code/code-dot-org/lib/dynamic_config/loaders/dcdo_loader.rb:52:in `open'
    from /home/will/Code/code-dot-org/lib/dynamic_config/loaders/dcdo_loader.rb:52:in `load'
    from /home/will/Code/code-dot-org/bin/dynamic_config/lib/update_dcdo.rb:8:in `<main>'
```

**Comment by trevorb** [comment]
Yep, one of the things I'm going to do this week is rework the way we update these settings and most of the stuff in /bin is going to get thrown away.

**Comment by wjordan** [comment]
@trevorb I've added some comments to this commit, perhaps this is still WIP stuff in which case feel free to ignore.

one more note- in Ruby the `bin` folder is conventionally for [binstubs](https://github.com/sstephenson/rbenv/wiki/Understanding-binstubs) (though I know our /bin is bloated with all sorts of other executable scripts as well), in any case perhaps `dashboard/config` be a more appropriate path to keep these yaml files?

**Comment by wjordan** [comment]
ah ok, sounds good nevermind then :+1:

**Pull request merged**